### PR TITLE
fix tests after registration migration & rename meta-reference -> basic / llm_as_judge provider

### PR DESCRIPTION
**Pull request by yanxi0830**
# What does this PR do?

- fix broken tests after registration migration
- move meta-reference scoring into basic / llm_as_judge provider

## Feature/Issue validation/testing/test plan

```
pytest -v -s -m meta_reference_scoring_together_inference scoring/test_scoring.py 
pytest -v -s -m braintrust_scoring_together_inference scoring/test_scoring.py
pytest -v -s -m llm_as_judge_scoring_together_inference scoring/test_scoring.py
```

## Before submitting

- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Ran pre-commit to handle lint / formatting issues.
- [x] Read the [contributor guideline](https://github.com/meta-llama/llama-stack/blob/main/CONTRIBUTING.md),
      Pull Request section?
- [x] Updated relevant documentation.
- [x] Wrote necessary unit or integration tests.

### CODE REVIEW

**Review by raghotham — COMMENTED** [review]
are fn_defs still needed?

**Comment by yanxi0830** [comment]
> are fn_defs still needed?

Yes, since we want to pre-register the scoring functions, we are keeping them in code.

File: llama_stack/providers/inline/scoring/llm_as_judge/scoring.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    ) -> None:
+        self.config = config
+        self.datasetio_api = datasetio_api
+        self.datasets_api = datasets_api
+        self.inference_api = inference_api
+        self.scoring_fn_id_impls = {}
+
+    async def initialize(self) -> None:
+        for x in LLM_JUDGE_FNS:
+            impl = x(inference_api=self.inference_api)
```
Comment (dltn) [inline llama_stack/providers/inline/scoring/llm_as_judge/scoring.py:44]: nit: fn?

File: llama_stack/providers/inline/scoring/llm_as_judge/scoring.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+        scoring_fn_defs_list = [
+            fn_def
+            for impl in self.scoring_fn_id_impls.values()
+            for fn_def in impl.get_supported_scoring_fn_defs()
+        ]
+
+        for f in scoring_fn_defs_list:
+            assert f.identifier.startswith(
+                "llm-as-judge"
+            ), "All meta-reference scoring fn must have identifier prefixed with 'meta-reference'! "
```
Comment (dltn) [inline llama_stack/providers/inline/scoring/llm_as_judge/scoring.py:61]: `meta-reference` =/= `llm-as-judge`?

**Review by dltn — APPROVED** [review]

**Pull request merged**